build.sbt

```diff
@@ -251,3 +264,26 @@ lazy val tests: Project = (project in file("tests"))
     asyncHttpClientFs2Backend,
     okhttpMonixBackend
   )
+
+// https://stackoverflow.com/questions/25766797/how-do-i-start-a-server-before-running-a-test-suite-in-sbt
+lazy val testServer: Project = project
```

Comment:
To be honest I'm not a big fan of putting anything except the absolutely necessary in the build - such as starting test servers. It always ends up being quite a fragile solution.

Instead, could we maybe just start the server in the `beforeAll()` method of a test and stop it in `afterAll()`? This would also make it easier to understand the test, as all information on how the test runs would be in the test class, not spread across the test class and the build.

Don't get me wrong - I think that it's a very good idea to spread the tests across the subprojects, but I wouldn't involve SBT in the mix :). We could also use a random port quite easily then, instead of using the same port for each project.

Reply:
This is the only real way to do it for JavaScript as the tests run in the browser. So rather than having this only apply to JS I have just made it apply to both. It also has the benefit of removing akka libraries off the classpath when running tests and only needed to start a single server for all tests. I have a project that does start the server on the JVM in `beforeAll()` and via `sbt-revolver` for JS (it uses wiremock) but I think its nicer to have them both do the same thing unless you would prefer otherwise?

Reply:
Running tests in the browser seems rather heavy - isn't there another way? After all, we are testing APIs which don't require the window or use the DOM. 

Wouldn't https://github.com/scala-js/scala-js-env-phantomjs or https://github.com/scala-js/scala-js-env-jsdom-nodejs work?

Reply:
Phantomjs is no longer maintained and is replaced by headless Chrome. jsdom does not have support for fetch (jsdom issue 1724). It is not that heavy apart from having to have Chrome installed; Chrome starts fast and the tests only take ~5 seconds according to travis (it takes longer to compile the JavaScript test file). Also only the `FetchBackendTests` run against the browser, the other unit tests run on nodejs. Even if you could use nodejs you would need to use a http server that could compile to JavaScript.

Reply:
Ok, I see, thanks for the clarification. 

I understand why running all tests in the same manner (with the SBT setup) would be in some ways more elegant and nicer, but I still think that the benefits of having "self-contained" tests, that is such which fully define what they need, are big enough to justify having SBT-managed tests only for the JS version.

For example, a simple thing such as running a backend's tests from IntelliJ will be harder once the tests require the http server to be running. And that might be quite a common task when developing a non-JS backend.

Hence if you would be able to do it, I'd prefer to keep the sbt-version only for the JS backend. Is that possible? :)

Reply:
Done